### DISCUSSION TITLE
Is it possible to have more specific rules for associating files with a LS?

### DISCUSSION THREAD

**Discussion by ahmedneilhussain** [eclipse/lsp4e#412] [General]
For our LS, as well as the core source files, we have a project-level `.toml` file, analogous to `Cargo.toml` in Rust or `pom.xml` in a Maven java project, controlling project-wide language and codegen settings and so on.

We'd like to add code completion support for these `.toml` files to our LS, but we don't want arbitrary `.toml` files binding to our LS, particularly `Cargo.toml` ones!

I have found https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Freference%2Fref-content-type.htm that content types can key off file names/patterns rather than general file types, but I can't find any examples.

Anyone tried this with LSP4e - does it Just Work?

**Comment by mickaelistria** [comment]
Yes, that works; instead of adding filters in LSP4E enablement, it's often more efficient to properly detect and filter content-types.
See for instance
* https://github.com/eclipse/corrosion/blob/master/org.eclipse.corrosion/plugin.xml#L22
* https://github.com/eclipse-m2e/m2e-core/blob/master/org.eclipse.m2e.core/plugin.xml#L33
* https://github.com/eclipse/shellwax/blob/master/org.eclipse.shellwax.core/plugin.xml#L17

**Reply by ahmedneilhussain** [reply]
Thanks v much
